### ISSUE TITLE
Wishlist: additionalProperties restrictions on object values

### ISSUE THREAD

**Issue by tv42** [omissis/go-jsonschema#15]
It seems `additionalProperties` are ignored.

```
	"headers": {
	    "type": "object",
	    "properties": {},
	    "additionalProperties": {
		"type": "string"
	    }
	}
```

translates to

```
type FooHeaders map[string]interface{}
```

where I would really like to have a `map[string]string` for that use case.

Similarly, I would want a `map[string][]string` elsewhere.

Use case: HTTP etc headers. I know for a fact the values will only ever be strings.

**Comment by ziposcar** [comment]
Same wish

**Comment by atombender** [comment]
Closed by #18, I believe.
